cli.js

```diff
@@ -100,15 +100,14 @@ var api = new Api(cli.input.length ? cli.input : arrify(conf.files), {
 	cacheEnabled: cli.flags.cache !== false
 });
 
-var logger = new Logger();
-logger.api = api;
+var logger;
 
 if (cli.flags.tap) {
-	logger.use(tapReporter());
+	logger = new Logger(tapReporter(api));
```

Comment:
Also, if reporters are created using `new`, the check on their constructors (e.g. https://github.com/sindresorhus/ava/blob/master/lib/reporters/tap.js#L14-L16) can just throw if used without `new`, as other modules do.